tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -382,7 +385,7 @@ static void test_single_shot_alarm_instance(const struct device *dev, bool set_t
 	ticks = counter_us_to_ticks(dev, counter_period_us);
 	top_cfg.ticks = ticks;
 
-	cntr_alarm_cfg.flags = 0;
+	cntr_alarm_cfg.flags = COUNTER_ALARM_CFG_ABSOLUTE;
```

Comment:
With this update, it caused tests with twister-build failed:

```
INFO    - 102/651 nrf52_bsim                tests/drivers/counter/counter_basic_api/drivers.counter.basic_api.nrf_zli  FAILED Failed (native 0.195s)
...
INFO    - /__w/zephyr/zephyr/twister-out/nrf52_bsim/tests/drivers/counter/counter_basic_api/drivers.counter.basic_api.nrf_zli/handler.log
INFO    - 103/651 nrf52_bsim                tests/drivers/counter/counter_basic_api/drivers.counter.basic_api  FAILED Failed (native 0.318s)
INFO    - /__w/zephyr/zephyr/twister-out/nrf52_bsim/tests/drivers/counter/counter_basic_api/drivers.counter.basic_api/handler.log
...
```
If the update for test case is correct, maybe need the driver to check the failure issue.